core/async/src/test_loop.rs

```diff
@@ -104,6 +104,9 @@ pub struct TestLoopV2 {
     /// Shutdown flag. When this flag is true, delayed action runners will no
     /// longer post any new events to the event loop.
     shutting_down: Arc<AtomicBool>,
+    /// If present, a function to call to print something every time an event is
+    /// handled. Intended only for debugging.
+    every_event_callback: Option<Box<dyn FnMut(&TestLoopData)>>,
```

Comment:
Honestly a bit wary about adding something like this... 😅 but if it's required, we can keep it.